imod/mf6/dis.py

```diff
@@ -151,3 +160,81 @@ def _validate(self, schemata, **kwargs):
 
     def get_regrid_methods(self) -> Optional[dict[str, Tuple[RegridderType, str]]]:
         return self._regrid_method
+
+    @classmethod
+    @standard_log_decorator()
+    def from_imod5_data(
+        cls,
+        imod5_data: dict[str, dict[str, GridDataArray]],
+        regridder_types: Optional[dict[str, tuple[RegridderType, str]]] = None,
+    ) -> "StructuredDiscretization":
+        """
+        Construct package from iMOD5 data, loaded with the
+        :func:`imod.formats.prj.open_projectfile_data` function.
+
+        Method regrids all variables to a target grid with the smallest extent
+        and smallest cellsize available in all the grids. Consequently it
+        converts iMODFLOW data to MODFLOW 6 data.
+
+        .. note::
+
+            The method expects the iMOD5 model to be fully 3D, not quasi-3D.
+
+        Parameters
+        ----------
+        imod5_data: dict
+            Dictionary with iMOD5 data. This can be constructed from the
+            :func:`imod.formats.prj.open_projectfile_data` method.
+        regridder_types: dict, optional
+            Optional dictionary with regridder types for a specific variable.
+            Use this to override default regridding methods.
+
+        Returns
+        -------
+        Modflow 6 StructuredDiscretization package.
+
+        """
+        data = {
+            "idomain": imod5_data["bnd"]["ibound"].astype(np.int32),
+            "top": imod5_data["top"]["top"],
+            "bottom": imod5_data["bot"]["bottom"],
+        }
+        # TODO: Add validation
+        # It is possible that top and bottom grids are on different grids, then
+        # this doesn't work:
+        # assert np.all(imod5_data["top"]["top"][1:].data == imod5_data["bot"]["bottom"][:-1].data)
+        # assert np.isin(np.unique(imod5_data["bnd"]["ibound"]), [-1, 0, 1])
+
+        target_grid = get_smallest_target_grid(*data.values())
+
+        regridder_settings = cls._regrid_method
```

Comment:
make a deepcopy of cls._regrid_,method to avoid changing the defaults in this assignment. This is something that must be done at more locations. probably every implementation of "get_regrid_methods"should do this.

Reply:
Very good catch! Implementing this elsewhere is worthy of its own issue. Created an issue for this https://github.com/Deltares/imod-python/issues/1000